pig-runtime/src/org/partiql/pig/runtime/DomainVisitorTransformBase.kt

```diff
@@ -26,23 +26,39 @@ import com.amazon.ionelement.api.MetaContainer
 abstract class DomainVisitorTransformBase {
     open fun transformMetas(metas: MetaContainer) = metas
 
-    open fun transformAnyElement(node: AnyElement): AnyElement =
-        // TODO:  remove .asAnyElement() below when https://github.com/amzn/ion-element-kotlin/issues/36 is fixed.
-        node.copy(metas = transformMetas(node.metas)).asAnyElement()
+    open fun transformAnyElement(node: AnyElement): AnyElement {
+        val newMetas = transformMetas(node.metas)
+        return if(node.metas !== newMetas) {
+            // TODO:  remove .asAnyElement() below when https://github.com/amzn/ion-element-kotlin/issues/36 is fixed.
```

Comment:
As a side note, is this TODO still relevant? The mentioned [issue](https://github.com/amzn/ion-element-kotlin/issues/36) was closed via:
> Actually, this is not possible since the narrower IonElement implementations already has a covariant return type which is narrowed.